docs/create-packages/native-files-in-net-packages.md

```diff
@@ -0,0 +1,165 @@
+runtimes/osx-arm64/lib/net8.0/Contoso.Native.dll
+runtimes/osx-arm64/native/libcontoso.dylib
+runtimes/osx-x64/lib/net8.0/Contoso.Native.dll
+runtimes/osx-x64/native/libcontoso.dylib
+runtimes/win-arm64/lib/net8.0/Contoso.Native.dll
+runtimes/win-arm64/native/contoso.dll
+runtimes/win-x64/lib/net8.0/Contoso.Native.dll
+runtimes/win-x64/native/contoso.dll
+```
+
+## Projects Targeting .NET Framework
+
+.NET has evolved over time, causing project and package incompatibility when certain features are used.
+What was described above assumes that the project using the package targets .NET 5 or higher, or .NET Core (`netcoreapp`), which uses the maximum features to simplify the package contents as much as possible.
+Projects targeting .NET Standard are not particularly relevant to this document, as those projects can only be class libraries, which cannot be run directly.
+However, projects targeting .NET Framework can use either SDK style or non-SDK style projects, and non-SDK style projects may use either `PackageReference` or `packages.config` to use NuGet packages.
+All of these combinations have different levels of support, and impose constraints on package authors who wish to support .NET Framework projects using the package.
+
+Our recommendation is to split the package into 3 packages.
+The first is a meta-package (a package with no contents/assets, only dependencies to other packages), that has dependencies to one of the other two packages, depending on target framework.
+The second package contains the .NET 5+ `ref/*` and `runtimes/*` assets as described above.
+The last package will contain the .NET Framework assets, and will not use the `ref/*` or `runtime/*` conventions.
+This is due to the build system having (partial) support for these NuGet conventions, but the runtime working sufficiently differently that it's easier to avoid it, to maximize project compatibility.
+
+The package with .NET Framework native libraries should have custom [MSBuild targets and props files](../concepts/MSBuild-props-and-targets.md), under `buildTransitive/<tfm>/<package ID>.[props|targets]`.
+For example, `buildTransitive/net472/Contoso.Native.targets`.
+To support `packages.config` projects, also have a `build/<tfm>/<package ID>.[props|targets]` which imports the `buildTransitive` file.
+This MSBuild script script should copy the native assemblies to subdirectories.
+
+Finally, the managed library that uses P/Invoke to call into the native library, it should have runtime checks to determine what platform the current executable is running in (x86, x64, or ARM64), and then P/Invoke into the native assembly in a correct directory.
+This is because .NET Framework supports the concept of "Any CPU" executables, which will execute (JIT compile) as x86 on x86 processors, and x64 otherwise (x64 or ARM64 processors).
+.NET Core removed this "Any CPU" executable concept, and separate x64, x86, and ARM64 executables need to be compiled and deployed explicitly.
```

Comment:
.NET Core does still have the concept of an `AnyCPU` app.  The app DLL can be architecture neutral, and can be run using the `dotnet` host, for example `dotnet app.dll`.  In this case the RID-specific native assets will be put in subfolders of the output directory, and the .NET host will set it up so that PInvoke calls will resolve to the right one.  This is what enables .NET Core apps to not need to do conditional PInvokes based on the architecture (as long as the API being called is the same across architectures).

Reply:
I've updated the wording. Please provide another comment if you think anything is inaccurate or could be worded better.